muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/chat/dao/UserChatSettingsDAO.java

```diff
@@ -0,0 +1,68 @@
+package fi.otavanopisto.muikku.plugins.chat.dao;
+
+import java.util.List;
+
+import javax.persistence.EntityManager;
+import javax.persistence.TypedQuery;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+
+import fi.otavanopisto.muikku.plugins.CorePluginsDAO;
+import fi.otavanopisto.muikku.plugins.chat.model.UserChatSettings;
+import fi.otavanopisto.muikku.plugins.chat.model.UserChatSettings_;
+import fi.otavanopisto.muikku.plugins.chat.model.UserChatVisibility;
+import fi.otavanopisto.muikku.plugins.forum.model.ForumMessage;
+import fi.otavanopisto.muikku.schooldata.SchoolDataIdentifier;
+
+
+public class UserChatSettingsDAO extends CorePluginsDAO<UserChatSettings> {
+	
+  private static final long serialVersionUID = -7830619828801454118L;
+
+  public UserChatSettings create(String userIdentifier, UserChatVisibility userChatVisibility) {
+    UserChatSettings settings = new UserChatSettings(userIdentifier, userChatVisibility);
+    getEntityManager().persist(settings);
+    return settings;
+  }
+  
+  public List<UserChatSettings> listAll(String userIdentifier, UserChatVisibility userChatVisibility) {
+	EntityManager entityManager = getEntityManager(); 
+	    
+	CriteriaBuilder criteriaBuilder = entityManager.getCriteriaBuilder();
+	CriteriaQuery<UserChatSettings> criteria = criteriaBuilder.createQuery(UserChatSettings.class);
+	Root<UserChatSettings> root = criteria.from(UserChatSettings.class);
+	criteria.select(root);
+	
+	TypedQuery<UserChatSettings> query = entityManager.createQuery(criteria);
+	
+	return query.getResultList();
+  }
+  
+  public UserChatSettings findByUser(SchoolDataIdentifier userIdentifier) {
```

Comment:
We try to keep DAO methods as simple and closely resembling the entity class as possible. As such, this method would be better as...

`public UserChatSettings findByUserIdentifier(String userIdentifier)`

...so there's less chance of getting confused whether the method deals with Users, SchoolDataIdentifiers, or a plain String (which it does as the `userIdentifer` in the `UserChatSettings` entity is a string).

Reply:
I sort of disagree with the comment because String as parameter is kind of obscure in itself and often having typed parameters helps give correct parameters and enforce type safety via compiler (more of a case when you have f.ex. userEntityId, the DAO method can accept UserEntity as parameter and use UserEntity.getId(), like it's done here).

In this case though the SchoolDataIdentifier is also kind of obscure so I'm fine with w/e.

Reply:
True dat. Maybe enough to rename method to findByUserIdentifier and leave the argument as-is.